src/main/java/edu/cornell/kfs/coa/businessobject/GlobalObjectWithIndirectCostRecoveryAccounts.java

```diff
@@ -0,0 +1,109 @@
+package edu.cornell.kfs.coa.businessobject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.kuali.kfs.coa.businessobject.IndirectCostRecoveryAccount;
+import org.kuali.rice.krad.bo.GlobalBusinessObjectDetailBase;
+import org.kuali.rice.krad.bo.PersistableBusinessObjectBase;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+public abstract class GlobalObjectWithIndirectCostRecoveryAccounts extends PersistableBusinessObjectBase {
```

Comment:
If this is only intended for use by global BOs, I would suggest making this class implement GlobalBusinessObject to avoid confusion. Since this is an abstract class, you don't have to specifically define the global BO methods here.